src/webservice/Router.h

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#pragma once
+
+#include <memory>
+#include <regex>
+#include <unordered_map>
+#include <vector>
+
+#include <proxygen/lib/http/HTTPMethod.h>
+
+#include "common/cpp/helpers.h"
+
+namespace proxygen {
+class RequestHandler;
+class HTTPMessage;
+}   // namespace proxygen
+
+namespace nebula {
+namespace web {
+
+using PathParams = std::unordered_map<std::string, std::string>;
+using ReqHandlerGenerator = std::function<proxygen::RequestHandler *(PathParams &&)>;
+
+class Route final {
+public:
+    Route(proxygen::HTTPMethod method, const std::string &path) : method_(method) {
+        checkPath(path);
+        setPath(path);
+    }
+
+    void setNext(Route *next) {
+        next_ = next;
+    }
+
+    Route *next() const {
+        return next_;
+    }
+
+    bool matches(proxygen::HTTPMethod method, const std::string &path) const;
+
+    Route &handler(ReqHandlerGenerator generator) {
+        this->generator_ = generator;
```

Comment:
Is here support to modify handler multi times? If not, need to judge the reentrant

Reply:
I want to manage all handlers chain later, so here this generator will be replaced by the latest. I can add some comments and do some checks for that.